server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -73,6 +73,12 @@ object LivyConf {
   val KINIT_FAIL_THRESHOLD =
     LivyConf.Entry("livy.server.launch.kerberos.kinit_fail_threshold", 5)
 
+  // If Livy can't find the yarn app within this time, consider it lost.
+  val YARN_APP_LOOKUP_TIMEOUT = Entry("livy.server.yarn.app-lookup-timeout", "30s")
+
+  // How often Livy polls YARN to refresh YARN app state.
+  val YARN_POLL_INTERVAL = Entry("livy.server.yarn.poll-interval", "1s")
```

Comment:
`1s` seems pretty aggressive, especially since the heartbeat time for AMs is generally much higher than that (Spark is 5s by default IIRC). Might be good to tune this down, especially since currently you have one thread per session doing the polling.